pkgs/tools/graphics/zbar/default.nix

```diff
@@ -1,43 +1,76 @@
-{ stdenv, fetchFromGitHub, imagemagickBig, pkgconfig, python2Packages, perl
-, libX11, libv4l, qt5, gtk2, xmlto, docbook_xsl, autoreconfHook, dbus
-, enableVideo ? stdenv.isLinux, enableDbus ? stdenv.isLinux
+{ stdenv
+, lib
+, fetchFromGitHub
+, imagemagickBig
+, pkgconfig
+, libX11
+, libv4l
+, qt5
+, gtk3
+, xmlto
+, docbook_xsl
+, autoreconfHook
+, dbus
+, enableVideo ? stdenv.isLinux
+, enableDbus ? stdenv.isLinux
 }:
 
-with stdenv.lib;
-let
-  inherit (python2Packages) pygtk python;
-in stdenv.mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "zbar";
   version = "0.23";
 
+  outputs = [ "out" "lib" "dev" "doc" "man" ];
+
   src = fetchFromGitHub {
     owner = "mchehab";
     repo = "zbar";
     rev = version;
     sha256 = "0hlxakpyjg4q9hp7yp3har1n78341b4knwyll28hn48vykg28pza";
   };
 
-  nativeBuildInputs = [ pkgconfig xmlto autoreconfHook docbook_xsl ];
+  nativeBuildInputs = [
+    pkgconfig
+    xmlto
+    autoreconfHook
+    docbook_xsl
+    qt5.wrapQtAppsHook
+  ];
 
   buildInputs = [
-    imagemagickBig python pygtk perl libX11
-  ] ++ optional enableDbus dbus
-  ++ optionals enableVideo [
-    libv4l gtk2 qt5.qtbase qt5.qtx11extras
+    imagemagickBig
+    libX11
+  ] ++ lib.optionals enableDbus [
+    dbus
+  ] ++ lib.optionals enableVideo [
+    libv4l
+    gtk3
+    qt5.qtbase
```

Comment:
Shouldn't this be called with `libsForQt5.callPackage` and then use the qt packages in scope? (no qt5.*)